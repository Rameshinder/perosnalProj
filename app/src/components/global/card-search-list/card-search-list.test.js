import React from 'react'
import { shallow, mount } from 'enzyme'

import CardSearchList from './index'
import CardListJson from '../stubs/cardList.json'

jest.mock('../../icon', () => 'Icon')

jest.mock('@anz/tooltip', () => 'Tooltip')
jest.mock('@anz/tooltip', () => 'TooltipWrapper')
jest.mock('@anz/icon/line/arrows-and-symbols/information', () => 'InfoIcon')

describe('Card', () => {
  const app = shallow(<CardSearchList cardlist={CardListJson} />)

  test('renders properly', () => {
    expect(app).toMatchSnapshot()
  })

  it('Card should render without crashing', () => {
    shallow(<CardSearchList cardlist={CardListJson} />)
  })

  it('Card should render without crashing', () => {
    mount(<CardSearchList cardlist={CardListJson.cards} />)
  })

  it('should filter cards when card name or number is input for search', () => {
    const spy = jest.spyOn(CardSearchList.prototype, 'onInputChange')
    const wrapper = mount(<CardSearchList cardlist={CardListJson.cards} />)
    const inputSearch = wrapper.find('input[name="card-search-input"]')
    wrapper.setState({ searchTerm: 'card' })
    inputSearch.simulate('change')
    expect(spy).toHaveBeenCalled()
  })

  it('should show cards when card name or number is input for search and matches card name', () => {
    const spy = jest.spyOn(CardSearchList.prototype, 'onInputChange')
    const wrapper = mount(<CardSearchList cardlist={CardListJson} />)
    const inputSearch = wrapper.find('input[name="card-search-input"]')
    wrapper.setState({ searchTerm: '4072200010054642' })
    inputSearch.simulate('change')
    expect(spy).toHaveBeenCalled()
  })

  it('should have not filter cards when card name or number does not match', () => {
    const spy = jest.spyOn(CardSearchList.prototype, 'onInputChange')
    const wrapper = mount(<CardSearchList cardlist={CardListJson.cards} />)
    const inputSearch = wrapper.find('input[name="card-search-input"]')
    wrapper.setState({ searchTerm: 'xyz' })
    inputSearch.simulate('change')
    expect(spy).toHaveBeenCalled()
  })

  it('should should show all cards when card name or number is empty or undefined', () => {
    const spy = jest.spyOn(CardSearchList.prototype, 'onInputChange')
    const wrapper = mount(<CardSearchList cardlist={CardListJson.cards} />)
    const inputSearch = wrapper.find('input[name="card-search-input"]')
    wrapper.setState({ searchTerm: '' })
    inputSearch.simulate('change')
    expect(spy).toHaveBeenCalled()
  })

  it('should clear the search field when cross icon is clicked', () => {
    const spy = jest.spyOn(CardSearchList.prototype, 'clearSearchInput')
    const wrapper = mount(<CardSearchList cardlist={CardListJson.cards} handleCardSelection={jest.fn} />)
    const inputSearch = wrapper.find('input[name="card-search-input"]')
    wrapper.setState({ searchTerm: 'card' })
    inputSearch.simulate('change')
    const clearSearch = wrapper.find('a[id="clear-search"]')
    clearSearch.simulate('click')
    expect(spy).toHaveBeenCalled()
  })
})
