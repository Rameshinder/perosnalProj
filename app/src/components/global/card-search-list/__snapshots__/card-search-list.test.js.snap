// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardSearchList
    cardlist={
      Object {
        "cards": Array [
          Object {
            "activationRequired": false,
            "cardSetID": "4072200010054642",
            "category": "CREDIT",
            "designCode": "0",
            "eligibleFor": Array [
              "AppleWalletProvisioning",
              "CardActivation",
              "SetPin",
              "ShowTC",
            ],
            "name": "MR PATRICK GLEN WILKINS",
            "number": "4072200010054642",
            "primary": true,
            "productCode": "PCP",
            "productSubCode": "008",
            "reissued": false,
            "status": "",
            "type": "ANZ Frequent Flyer Platinum Visa",
          },
          Object {
            "activationRequired": false,
            "cardSetID": "4622397029822304",
            "category": "DEBIT",
            "designCode": "0",
            "eligibleFor": Array [
              "CardActivation",
              "ShowTC",
              "SetPin",
            ],
            "name": "MR PATRICK GLEN WILKINS",
            "number": "4622397029822304",
            "primary": true,
            "productCode": "PCP",
            "productSubCode": "061",
            "reissued": false,
            "status": "",
            "type": "ANZ Frequent Flyer Platinum Visa",
          },
          Object {
            "activationRequired": false,
            "cardSetID": "5432397029822001",
            "category": "CREDIT",
            "designCode": "0",
            "eligibleFor": Array [
              "CardActivation",
              "SetPin",
              "ShowTC",
            ],
            "name": "MR PATRICK GLEN WILKINS",
            "number": "5432397029822001",
            "primary": true,
            "productCode": "PCP",
            "productSubCode": "061",
            "reissued": false,
            "status": "",
            "type": "ANZ Frequent Flyer Platinum Master Card",
          },
        ],
        "cardsNonEligible": Array [
          Object {
            "activationRequired": false,
            "cardSetID": "4072200010054642",
            "category": "CREDIT",
            "designCode": "0",
            "eligibleFor": Array [
              "AppleWalletProvisioning",
              "CardActivation",
              "ShowTC",
            ],
            "name": "MR PATRICK GLEN WILKINS",
            "number": "4072200010054642",
            "primary": true,
            "productCode": "PCP",
            "productSubCode": "008",
            "reissued": false,
            "status": "",
            "type": "ANZ Frequent Flyer Platinum Visa",
          },
          Object {
            "activationRequired": false,
            "cardSetID": "4622397029822304",
            "category": "DEBIT",
            "designCode": "0",
            "eligibleFor": Array [
              "CardActivation",
              "ShowTC",
              "SetPin",
            ],
            "name": "MR PATRICK GLEN WILKINS",
            "number": "4622397029822304",
            "primary": true,
            "productCode": "PCP",
            "productSubCode": "061",
            "reissued": false,
            "status": "",
            "type": "ANZ Frequent Flyer Platinum Visa",
          },
          Object {
            "activationRequired": false,
            "cardSetID": "5432397029822001",
            "category": "CREDIT",
            "designCode": "0",
            "eligibleFor": Array [
              "CardActivation",
              "ShowTC",
            ],
            "name": "MR PATRICK GLEN WILKINS",
            "number": "5432397029822001",
            "primary": true,
            "productCode": "PCP",
            "productSubCode": "061",
            "reissued": false,
            "status": "",
            "type": "ANZ Frequent Flyer Platinum Master Card",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.h3
            data-test-id="card-Search-Heading"
            proLight={true}
          >
            Select an eligible card below
          </styled.h3>
        </styled.div>,
        <styled.div>
          <styled.a>
            <Icon
              iconName="Search"
            />
          </styled.a>
          <InputText
            ariaLabel="search text input"
            ariaRequired={false}
            autocapsformat={false}
            data-test-id="card-search-input"
            id="card-search-input"
            inputValue=""
            name="card-search-input"
            onChange={[Function]}
            placeholderText="Filter by card name or number"
          />
        </styled.div>,
        <CardList
          cards={
            Array [
              Object {
                "activationRequired": false,
                "cardSetID": "4072200010054642",
                "category": "CREDIT",
                "designCode": "0",
                "eligibleFor": Array [
                  "AppleWalletProvisioning",
                  "CardActivation",
                  "SetPin",
                  "ShowTC",
                ],
                "name": "MR PATRICK GLEN WILKINS",
                "number": "4072200010054642",
                "primary": true,
                "productCode": "PCP",
                "productSubCode": "008",
                "reissued": false,
                "status": "",
                "type": "ANZ Frequent Flyer Platinum Visa",
              },
              Object {
                "activationRequired": false,
                "cardSetID": "4622397029822304",
                "category": "DEBIT",
                "designCode": "0",
                "eligibleFor": Array [
                  "CardActivation",
                  "ShowTC",
                  "SetPin",
                ],
                "name": "MR PATRICK GLEN WILKINS",
                "number": "4622397029822304",
                "primary": true,
                "productCode": "PCP",
                "productSubCode": "061",
                "reissued": false,
                "status": "",
                "type": "ANZ Frequent Flyer Platinum Visa",
              },
              Object {
                "activationRequired": false,
                "cardSetID": "5432397029822001",
                "category": "CREDIT",
                "designCode": "0",
                "eligibleFor": Array [
                  "CardActivation",
                  "SetPin",
                  "ShowTC",
                ],
                "name": "MR PATRICK GLEN WILKINS",
                "number": "5432397029822001",
                "primary": true,
                "productCode": "PCP",
                "productSubCode": "061",
                "reissued": false,
                "status": "",
                "type": "ANZ Frequent Flyer Platinum Master Card",
              },
            ]
          }
          handleCardSelection={undefined}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.h3
            data-test-id="card-Search-Heading"
            proLight={true}
          >
            Select an eligible card below
          </styled.h3>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Select an eligible card below",
            "data-test-id": "card-Search-Heading",
            "proLight": true,
          },
          "ref": null,
          "rendered": "Select an eligible card below",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.a>
              <Icon
                iconName="Search"
              />
            </styled.a>,
            <InputText
              ariaLabel="search text input"
              ariaRequired={false}
              autocapsformat={false}
              data-test-id="card-search-input"
              id="card-search-input"
              inputValue=""
              name="card-search-input"
              onChange={[Function]}
              placeholderText="Filter by card name or number"
            />,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Icon
                iconName="Search"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "iconName": "Search",
              },
              "ref": null,
              "rendered": null,
              "type": "Icon",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "ariaLabel": "search text input",
              "ariaRequired": false,
              "autocapsformat": false,
              "data-test-id": "card-search-input",
              "id": "card-search-input",
              "inputValue": "",
              "name": "card-search-input",
              "onChange": [Function],
              "placeholderText": "Filter by card name or number",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cards": Array [
            Object {
              "activationRequired": false,
              "cardSetID": "4072200010054642",
              "category": "CREDIT",
              "designCode": "0",
              "eligibleFor": Array [
                "AppleWalletProvisioning",
                "CardActivation",
                "SetPin",
                "ShowTC",
              ],
              "name": "MR PATRICK GLEN WILKINS",
              "number": "4072200010054642",
              "primary": true,
              "productCode": "PCP",
              "productSubCode": "008",
              "reissued": false,
              "status": "",
              "type": "ANZ Frequent Flyer Platinum Visa",
            },
            Object {
              "activationRequired": false,
              "cardSetID": "4622397029822304",
              "category": "DEBIT",
              "designCode": "0",
              "eligibleFor": Array [
                "CardActivation",
                "ShowTC",
                "SetPin",
              ],
              "name": "MR PATRICK GLEN WILKINS",
              "number": "4622397029822304",
              "primary": true,
              "productCode": "PCP",
              "productSubCode": "061",
              "reissued": false,
              "status": "",
              "type": "ANZ Frequent Flyer Platinum Visa",
            },
            Object {
              "activationRequired": false,
              "cardSetID": "5432397029822001",
              "category": "CREDIT",
              "designCode": "0",
              "eligibleFor": Array [
                "CardActivation",
                "SetPin",
                "ShowTC",
              ],
              "name": "MR PATRICK GLEN WILKINS",
              "number": "5432397029822001",
              "primary": true,
              "productCode": "PCP",
              "productSubCode": "061",
              "reissued": false,
              "status": "",
              "type": "ANZ Frequent Flyer Platinum Master Card",
            },
          ],
          "handleCardSelection": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.h3
              data-test-id="card-Search-Heading"
              proLight={true}
            >
              Select an eligible card below
            </styled.h3>
          </styled.div>,
          <styled.div>
            <styled.a>
              <Icon
                iconName="Search"
              />
            </styled.a>
            <InputText
              ariaLabel="search text input"
              ariaRequired={false}
              autocapsformat={false}
              data-test-id="card-search-input"
              id="card-search-input"
              inputValue=""
              name="card-search-input"
              onChange={[Function]}
              placeholderText="Filter by card name or number"
            />
          </styled.div>,
          <CardList
            cards={
              Array [
                Object {
                  "activationRequired": false,
                  "cardSetID": "4072200010054642",
                  "category": "CREDIT",
                  "designCode": "0",
                  "eligibleFor": Array [
                    "AppleWalletProvisioning",
                    "CardActivation",
                    "SetPin",
                    "ShowTC",
                  ],
                  "name": "MR PATRICK GLEN WILKINS",
                  "number": "4072200010054642",
                  "primary": true,
                  "productCode": "PCP",
                  "productSubCode": "008",
                  "reissued": false,
                  "status": "",
                  "type": "ANZ Frequent Flyer Platinum Visa",
                },
                Object {
                  "activationRequired": false,
                  "cardSetID": "4622397029822304",
                  "category": "DEBIT",
                  "designCode": "0",
                  "eligibleFor": Array [
                    "CardActivation",
                    "ShowTC",
                    "SetPin",
                  ],
                  "name": "MR PATRICK GLEN WILKINS",
                  "number": "4622397029822304",
                  "primary": true,
                  "productCode": "PCP",
                  "productSubCode": "061",
                  "reissued": false,
                  "status": "",
                  "type": "ANZ Frequent Flyer Platinum Visa",
                },
                Object {
                  "activationRequired": false,
                  "cardSetID": "5432397029822001",
                  "category": "CREDIT",
                  "designCode": "0",
                  "eligibleFor": Array [
                    "CardActivation",
                    "SetPin",
                    "ShowTC",
                  ],
                  "name": "MR PATRICK GLEN WILKINS",
                  "number": "5432397029822001",
                  "primary": true,
                  "productCode": "PCP",
                  "productSubCode": "061",
                  "reissued": false,
                  "status": "",
                  "type": "ANZ Frequent Flyer Platinum Master Card",
                },
              ]
            }
            handleCardSelection={undefined}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.h3
              data-test-id="card-Search-Heading"
              proLight={true}
            >
              Select an eligible card below
            </styled.h3>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Select an eligible card below",
              "data-test-id": "card-Search-Heading",
              "proLight": true,
            },
            "ref": null,
            "rendered": "Select an eligible card below",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.a>
                <Icon
                  iconName="Search"
                />
              </styled.a>,
              <InputText
                ariaLabel="search text input"
                ariaRequired={false}
                autocapsformat={false}
                data-test-id="card-search-input"
                id="card-search-input"
                inputValue=""
                name="card-search-input"
                onChange={[Function]}
                placeholderText="Filter by card name or number"
              />,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Icon
                  iconName="Search"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "iconName": "Search",
                },
                "ref": null,
                "rendered": null,
                "type": "Icon",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "ariaLabel": "search text input",
                "ariaRequired": false,
                "autocapsformat": false,
                "data-test-id": "card-search-input",
                "id": "card-search-input",
                "inputValue": "",
                "name": "card-search-input",
                "onChange": [Function],
                "placeholderText": "Filter by card name or number",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            null,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cards": Array [
              Object {
                "activationRequired": false,
                "cardSetID": "4072200010054642",
                "category": "CREDIT",
                "designCode": "0",
                "eligibleFor": Array [
                  "AppleWalletProvisioning",
                  "CardActivation",
                  "SetPin",
                  "ShowTC",
                ],
                "name": "MR PATRICK GLEN WILKINS",
                "number": "4072200010054642",
                "primary": true,
                "productCode": "PCP",
                "productSubCode": "008",
                "reissued": false,
                "status": "",
                "type": "ANZ Frequent Flyer Platinum Visa",
              },
              Object {
                "activationRequired": false,
                "cardSetID": "4622397029822304",
                "category": "DEBIT",
                "designCode": "0",
                "eligibleFor": Array [
                  "CardActivation",
                  "ShowTC",
                  "SetPin",
                ],
                "name": "MR PATRICK GLEN WILKINS",
                "number": "4622397029822304",
                "primary": true,
                "productCode": "PCP",
                "productSubCode": "061",
                "reissued": false,
                "status": "",
                "type": "ANZ Frequent Flyer Platinum Visa",
              },
              Object {
                "activationRequired": false,
                "cardSetID": "5432397029822001",
                "category": "CREDIT",
                "designCode": "0",
                "eligibleFor": Array [
                  "CardActivation",
                  "SetPin",
                  "ShowTC",
                ],
                "name": "MR PATRICK GLEN WILKINS",
                "number": "5432397029822001",
                "primary": true,
                "productCode": "PCP",
                "productSubCode": "061",
                "reissued": false,
                "status": "",
                "type": "ANZ Frequent Flyer Platinum Master Card",
              },
            ],
            "handleCardSelection": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
